editor/import/scene_importer_mesh.cpp

```diff
@@ -261,6 +266,103 @@ void EditorSceneImporterMesh::clear() {
 	mesh.unref();
 }
 
+void EditorSceneImporterMesh::create_shadow_mesh() {
+	if (shadow_mesh.is_valid()) {
+		shadow_mesh.unref();
+	}
+
+	shadow_mesh.instance();
+
+	//no shadow mesh for blendshapes
+	if (blend_shapes.size() > 0) {
+		return;
+	}
+	//no shadow mesh for skeletons
+	for (int i = 0; i < surfaces.size(); i++) {
+		if (surfaces[i].arrays[RS::ARRAY_BONES].get_type() != Variant::NIL) {
+			return;
+		}
+		if (surfaces[i].arrays[RS::ARRAY_WEIGHTS].get_type() != Variant::NIL) {
+			return;
+		}
+	}
+
+	for (int i = 0; i < surfaces.size(); i++) {
+		LocalVector<int> vertex_remap;
+		Vector<Vector3> new_vertices;
+		Vector<Vector3> vertices = surfaces[i].arrays[RS::ARRAY_VERTEX];
+		int vertex_count = vertices.size();
+		{
+			Map<Vector3, int> unique_vertices;
+			const Vector3 *vptr = vertices.ptr();
+			for (int j = 0; j < vertex_count; j++) {
+				Vector3 v = vptr[j];
+
+				Map<Vector3, int>::Element *E = unique_vertices.find(v);
+
+				if (E) {
+					vertex_remap.push_back(E->get());
+				} else {
+					int vcount = unique_vertices.size();
+					unique_vertices[v] = vcount;
+					vertex_remap.push_back(vcount);
+					new_vertices.push_back(v);
+				}
+			}
+		}
+
+		Array new_surface;
+		new_surface.resize(RS::ARRAY_MAX);
+		Dictionary lods;
+
+		print_line("original vertex count: " + itos(vertices.size()) + " new vertex count: " + itos(new_vertices.size()));
```

Comment:
Probably needs to be removed or changed to a verbose print.

Reply:
done